### DISCUSSION TITLE
Note3_Output_Additional_Variables

### DISCUSSION THREAD

**Discussion by Dylan-leo1** [NCAR/hrldas#234] [Q&A] [answered]
After following the tutorial, I also executed the make clean and make commands, but when I used ncview to view the single-station run results, I did not find the added variables BTRANXY and DB, in both examples.

**Comment by tslin2** [comment]
I am not sure what could be the issue, but you can go through again to double check if your code changes are made in *.f90 after compiling

**Reply by Dylan-leo1** [reply]
According to the tutorial, I modified the *.F90 file. I checked the *.f90 file and found that there were indeed no added variables. Do I need to modify the *.f90 file directly before compiling?

**Reply by cenlinhe** [reply]
the .f90 file is an intermediate file generated after compiling. After you modify the *.F90 file, you need to re-compile the entire model by doing the following:
first, "make clean" to clean up everything,
second, "make all" to re-compile.
If you do not "make clean", then even if you re-compile by "make all", your new changes will not be there.

**Reply by Dylan-leo1** [reply]
I have modified the *.F90 file and performed "make clean" and "make all", but after compiling, the *.f90 file still does not have the newly added variables.  Maybe when I executed “make clean”, it only deleted the hrldas.exe file and did not delete the *.f90 file. Do I need to delete *.f90 before compiling?

**Reply by cenlinhe** [reply, marked as answer]
no, "make clean" should delete all "*.f90, *.o, *.exe" files automatically. You need to issue the "make clean" command at the level of hrldas/hrldas folder not the noahmp folder.

**Reply by cenlinhe** [reply]
after you "make clean", you can double check if those *.f90 files are deleted or not. and then after confirming, you can do "make all".

**Reply by Dylan-leo1** [reply]
Thank you very much for your guidance. After executing the “make clean” and “make all”  in the hrldas/hrldas/ directory, it was indeed successful.
